hw/ip/flash_ctrl/doc/_index.md

```diff
@@ -222,6 +218,28 @@ The valid program range is thus the valid program resolution for a particular me
 
 This information is not configurable but instead decided at design time and is exposed as a readable status.
 
+#### Erase Suspend
+
+The flash controller supports erase suspend through {{< regref "ERASE_SUSPEND" >}}.
+This allows the software to interrupt an ongoing erase operation.
+
+The behavior of what happens to flash contents when erase is suspended is vendor defined; however, generally it can be assumed that the erase would be incomplete.
+It is then up to the controlling software to take appropriate steps to erase again at a later time.
+
+#### Additional Flash Attributes
+
+There are certain attributes provisioned in {{< regref "MP_REGION_CFG_SHADOWED_0" >}} that are not directly used by the open source protocol or physical controllers.
+
+Instead, these attributes are fed to the vendor flash module on a per-page or defined boundary basis.
+Currently there is only one such attribute {{< regref "MP_REGION_CFG_SHADOWED_0.HE" >}}.
+
+#### Idle Indication to External Power Manager
+
+The flash controller provides an idle indication to an external power manager.
+This idle indication does not mean the controller is doing "nothing", but rather the controller is not doing anything "stateful", ie program or erase.
+
+This is because an external power manager event (such as shutting off power) while a flash stateful transaction is ongoing may be damaging to the vendor flash module.
```

Comment:
Well, power can be shut off externally: in what state will flash end up if that happens? Can it become corrupted? And what about regular writes?

Reply:
so if the power is pulled from the outside, it actually activates the "brownout" mechanism from the AST.  Which basically informs the flash "holy crap the power is dying, you have to finish NOW!".

There will basically be timing requirements on how quickly the supply can discharge in order for this to to work.  But to your point basically.. this kind of sudden power pull when the device is ongoing could be dangerous.